addons/mrp_subcontracting_account/models/stock_move.py

```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    def _get_price_unit(self):
+        self.ensure_one()
+        if self.is_subcontract:
+            return self.price_unit
```

Comment:
There is an issue here. In case you have purchase and mrp_subcontracting installed. And _get_price_unit is called first here (there is no dependency between purchase_stock and  mrp_subcontracting_account). You will never read the price on the PO line.

An idea would be have a new hook called inside _get_price_unit of stock_account only relevant in mrp_suibcotracting_account